tests/Functional/Ticket/DBAL4283Test.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Ticket;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Platforms\SQLServer2012Platform;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SQLServerSchemaManager;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+/**
+ * Fixes DBAL issue #4283: Doctrine fails to declare Extended Properties on SQL-Server when table name is quoted.
+ *
+ * The problem is in the SQLServerPlatform which incorrectly quotes the SCHEMA and TABLE object names
+ * when they contain Reserved Words.
+ *
+ * The fix is to keep the SCHEMA and TABLE as identifiers in the T-SQL query.
+ *
+ * @link https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/basic-mapping.html#quoting-reserved-words
+ * @link https://github.com/doctrine/dbal/issues/4283
+ */
+class DBAL4283Test extends FunctionalTestCase
+{
+    public function testAddingExtendedPropertyToUnquotedTableName(): void
+    {
+        $createTableSQL = $this->getCreateTableSQL('table_name');
+
+        self::assertEquals(
+            "EXEC sp_addextendedproperty N'MS_Description', N'(DC2Type:array)', "
+            . "N'SCHEMA', 'dbo', N'TABLE', 'table_name', N'COLUMN', column_name",
+            $createTableSQL
+        );
+    }
+
+    public function testAddingExtendedPropertyToQuotedTableName(): void
+    {
+        $createTableSQL = $this->getCreateTableSQL('`quoted-table-name`');
+
+        self::assertEquals(
+            "EXEC sp_addextendedproperty N'MS_Description', N'(DC2Type:array)', "
+            . "N'SCHEMA', 'dbo', N'TABLE', [quoted-table-name], N'COLUMN', column_name",
+            $createTableSQL
+        );
+    }
+
+    public function testAddingExtendedPropertyToQuotedTableInSchema(): void
+    {
+        $createTableSQL = $this->getCreateTableSQL('`custom.quoted-table-name`');
+
+        self::assertEquals(
+            "EXEC sp_addextendedproperty N'MS_Description', N'(DC2Type:array)', "
+            . "N'SCHEMA', [custom], N'TABLE', [quoted-table-name], N'COLUMN', column_name",
+            $createTableSQL
+        );
+    }
+
+    public function getCreateTableSQL(string $quotedName): string
+    {
+        $driver   = $this->createMock(Driver::class);
```

Comment:
Since you perform no expectations, this is a stub:

```suggestion
        $driver   = $this->createStub(Driver::class);
```